src/main/java/io/cryostat/recordings/RecordingMetadataManager.java

```diff
@@ -153,12 +155,19 @@ public Metadata deleteRecordingMetadataIfExists(String targetId, String recordin
     public Map<String, String> parseRecordingLabels(String labels) throws IllegalArgumentException {
         Objects.requireNonNull(labels, "Labels must not be null");
 
+        Pattern noWhitespace = Pattern.compile("^(\\S)+$");
+        Matcher matcher = noWhitespace.matcher(labels);
```

Comment:
What is the value of `labels` at this point, or what is the source of the value? I think this is going to come in from ex. an API request, and it's the raw JSON representation of the labels map, right?

In that case it's probably better to apply the no-whitespace validation after the gson parser has already parsed it. We can be lenient about the JSON input being pretty-printed with whitespaces, it's just the actual values within that we want to enforce on.